### PR TITLE
Allowing search to be disabled for anonymous users. [Fixes #6473]

### LINKED ISSUE #6473: Anonymous access to search
Currently anonymous users can use the Search module, which could potentially be an issue. It would be nice if you could allow or disallow anonymous users access to search.

### PR DESCRIPTION
**Pull request by JRKelso**

### CODE REVIEW

File: src/Orchard.Web/Modules/Orchard.Search/Views/EditorTemplates/Parts/Search.SiteSettings.Fields.cshtml [hunk 1]
```diff
@@ -46,7 +46,11 @@
     </div>
     <div>
         <label for="@Html.FieldIdFor(m => m.DisplayType)">@T("Display Type")</label>
-        @Html.TextBoxFor(m => m.DisplayType, new { @class = "text single-line" })
+        @Html.TextBoxFor(m => m.DisplayType, new {@class = "text single-line"})
         <span class="hint">@T("The display type to use for content items in the search results page.")</span>
     </div>
+    <div>
+        <label>@T("Allow Anonymous Users to Search") @Html.EditorFor(m => m.AllowAnonymousSearch)</label>
+        <span class="hint">@T("If checked, search results will only be displayed to users who are authenticated.")</span>
```
Comment (paynecrl97) [inline src/Orchard.Web/Modules/Orchard.Search/Views/EditorTemplates/Parts/Search.SiteSettings.Fields.cshtml]: I think this hint text may be incorrect

**Comment by paynecrl97** [comment]
I may be wrong, but this looks like if it were merged as it is then `AllowAnonymousSearch` will default to false which will cause a breaking change to sites that already have search in place.

File: src/Orchard.Web/Modules/Orchard.Search/Views/EditorTemplates/Parts/Search.SiteSettings.Fields.cshtml [hunk 2]
```diff
@@ -46,7 +46,11 @@
...
         <label for="@Html.FieldIdFor(m => m.DisplayType)">@T("Display Type")</label>
-        @Html.TextBoxFor(m => m.DisplayType, new { @class = "text single-line" })
+        @Html.TextBoxFor(m => m.DisplayType, new {@class = "text single-line"})
         <span class="hint">@T("The display type to use for content items in the search results page.")</span>
     </div>
-</fieldset>
\ No newline at end of file
+    <div>
+        <label>@T("Allow Anonymous Users to Search") @Html.EditorFor(m => m.AllowAnonymousSearch)</label>
+        <span class="hint">@T("If unchecked, search results will only be displayed to users who are authenticated.")</span>
```
Comment (paynecrl97) [inline src/Orchard.Web/Modules/Orchard.Search/Views/EditorTemplates/Parts/Search.SiteSettings.Fields.cshtml]: FYI- if you add a class of `forcheckbox` to a label, the label will be positioned correctly for the checkbox eg:

```
	@Html.EditorFor(m => m.AllowAnonymousSearch)
	<label for="@Html.FieldIdFor(m => m.AllowAnonymousSearch)" class="forcheckbox">@T("...")</label>
```

Comment (JRKelso) [inline src/Orchard.Web/Modules/Orchard.Search/Views/EditorTemplates/Parts/Search.SiteSettings.Fields.cshtml]: Ah, thanks! I didn't know that, still quite new to C#, have barely scratched the surface of razor things.

**Comment by sebastienros** [comment]
Following the issue.

**Pull request closed without merging**